core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -42,20 +53,94 @@
 @Component
 @Order(value = SCMLinkValidationProbe.ORDER)
 public class SCMLinkValidationProbe extends Probe {
-    private static final Logger LOGGER = LoggerFactory.getLogger(SCMLinkValidationProbe.class);
-
-    private static final String GH_REGEXP = "https://(?<server>[^/]*)/(?<repo>jenkinsci/[^/]*)(?:/(?<folder>.*))?";
-    public static final Pattern GH_PATTERN = Pattern.compile(GH_REGEXP);
     public static final int ORDER = UpdateCenterPluginPublicationProbe.ORDER + 100;
     public static final String KEY = "scm";
+    private static final Logger LOGGER = LoggerFactory.getLogger(SCMLinkValidationProbe.class);
+    private static final String GH_REGEXP = "https://(?<server>[^/]*)/(?<repo>jenkinsci/[^/]*)";
+    public static final Pattern GH_PATTERN = Pattern.compile(GH_REGEXP);
 
     @Override
     public ProbeResult doApply(Plugin plugin, ProbeContext context) {
         if (plugin.getScm() == null || plugin.getScm().isBlank()) {
             LOGGER.warn("{} has no SCM link", plugin.getName());
-            return ProbeResult.error(key(), "The plugin SCM link is empty");
+            return ProbeResult.error(key(), "The plugin SCM link is empty.");
+        }
+        return fromSCMLink(context, plugin.getScm(), plugin.getName());
+    }
+
+    /**
+     * Validates the SCM link, and sets {@link ProbeContext#setScmFolderPath(Optional)}. The value is always the path of the POM file.
+     *
+     * @param context    Refer {@link ProbeContext}.
+     * @param scm        The SCM link {@link Plugin#getScm()}.
+     * @param pluginName The name of the plugin {@link Plugin#getName()}.
+     * @return ProbeResult {@link ProbeResult}.
+     */
+    private ProbeResult fromSCMLink(ProbeContext context, String scm, String pluginName) {
+        Matcher matcher = GH_PATTERN.matcher(scm);
+        if (!matcher.find()) {
+            LOGGER.atDebug().log(() -> String.format("%s is not respecting the SCM URL Template.", scm));
+            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories.");
+        }
+        try {
+            context.getGitHub().getRepository(matcher.group("repo"));
```

Comment:
That line is weird, cause a `get` method, often called a `getter`, is expected to return something, while here I guess it is just used in order to clone the repository right?

I don't expect you to change the code of the context object from there, that would be another refactoring, but a one line comment saying that this line clones the repository allows to understand why it is there.